core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsMetricsManagerImpl.java

```diff
@@ -0,0 +1,400 @@
+public class JGroupsMetricsManagerImpl implements JGroupsMetricsManager {
+
+   @Inject CacheManagerNotifier notifier;
+   @Inject MetricsRegistry registry;
+   @Inject TimeService timeService;
+
+   private final Map<Address, DestinationMetrics> perDestinationMetrics;
+   private final List<ClusterMetrics> otherChannels;
+   private final boolean histogramEnabled;
+   private volatile MainChannelRegistry mainChannelRegistry;
+   private volatile boolean stopped = true;
+
+   public JGroupsMetricsManagerImpl(boolean histogramEnabled) {
+      this.histogramEnabled = histogramEnabled;
+      perDestinationMetrics = new ConcurrentHashMap<>(16);
+      otherChannels = new CopyOnWriteArrayList<>();
+   }
+
+   @Start
+   public void start() {
+      stopped = false;
+      notifier.addListener(this);
+   }
+
+   @Stop
+   public void stop() {
+      stopped = true;
+      notifier.removeListener(this);
+      perDestinationMetrics.values().forEach(metrics -> metrics.unregister(registry));
+      perDestinationMetrics.clear();
+      otherChannels.forEach(metrics -> metrics.unregister(registry));
+      mainChannelRegistry = null;
+   }
+
+   @ViewChanged
+   @Merged
+   public void onViewChanged(ViewChangedEvent event) {
+      if (stopped) {
+         return;
+      }
+      var leftMembers = new HashSet<>(perDestinationMetrics.keySet());
+      event.getNewMembers().forEach(leftMembers::remove);
+
+      for (Address node : leftMembers) {
+         perDestinationMetrics.computeIfPresent(node, (unused, metrics) -> {
+            metrics.unregister(registry);
+            return null;
+         });
+      }
+   }
+
+   @Override
+   public RequestTracker trackRequest(Address destination) {
+      if (stopped) {
+         return new NoOpRequestTracker(destination);
+      }
+      var metrics = get(destination);
+      if (metrics == null) {
+         return new NoOpRequestTracker(destination);
+      }
+      return new RequestTrackerImpl(destination, metrics, timeService);
+   }
+
+   @Override
+   public void recordMessageSent(Address destination, int bytesSent, boolean async) {
+      if (stopped) {
+         return;
+      }
+      var metrics = get(destination);
+      if (metrics == null) {
+         return;
+      }
+      metrics.incrementBytesSent(bytesSent);
+      if (async) {
+         metrics.incrementAsyncRequests();
+      }
+   }
+
+   @Override
+   public synchronized void onChannelConnected(JChannel channel, boolean isMainChannel) {
+      if (stopped) {
+         return;
+      }
+      String nodeName = Objects.requireNonNull(nodeName(channel));
+      String clusterName = Objects.requireNonNull(channel.clusterName());
+      if (isMainChannel) {
+         assert mainChannelRegistry == null;
+         mainChannelRegistry = new MainChannelRegistry(nodeName, clusterName);
+      }
+      if (otherChannels.stream().map(m -> m.channel).noneMatch(ch -> ch.equals(channel))) {
+         otherChannels.add(new ClusterMetrics(channel));
```

Comment:
General question. In what scenario do we create additional JGroup's channels to be created?